tests/test_django.py

```diff
@@ -469,24 +470,28 @@ def test_with_content(self):
 
         # Django only allocates the full path on save()
         o.save()
-        self.assertEqual(b'foo', o.afile.read())
+        with o.afile as f:
+            self.assertEqual(b'foo', f.read())
         self.assertEqual('django/example.dat', o.afile.name)
 
     def test_with_file(self):
         with open(testdata.TESTFILE_PATH, 'rb') as f:
             o = WithFileFactory.build(afile__from_file=f)
             o.save()
 
-        self.assertEqual(b'example_data\n', o.afile.read())
+        with o.afile as f:
+            self.assertEqual(b'example_data\n', f.read())
         self.assertEqual('django/example.data', o.afile.name)
 
     def test_with_path(self):
         o = WithFileFactory.build(afile__from_path=testdata.TESTFILE_PATH)
         self.assertIsNone(o.pk)
 
-        # Django only allocates the full path on save()
-        o.save()
-        self.assertEqual(b'example_data\n', o.afile.read())
+        with o.afile:
+            # Django only allocates the full path on save()
+            o.save()
+        with o.afile as f:
```

Comment:
Rather than closing the file and opening it again, what do you think about using [`seek()`](https://docs.python.org/3.7/library/io.html#io.IOBase.seek). For example, we could do:

```python
with o.afile as f:
    # Django only allocates the full path on save()
    o.save()

    f.seek(0)
    self.assertEqual(b'example_data\n', f.read())
```